### DISCUSSION TITLE
useWaitForTransaction never errors

### DISCUSSION THREAD

**Discussion by isaac-martin** [wevm/wagmi#428] [Q&A]
I'm not sure if i am doing something wrong, but while using `useWaitForTransaction` and submitting a transaction that I expect to fail, I can never catch the error? I can see the error failing on rinkeby, but `useWaitForTransaction` always succeeds

Stripped back code below. 

```tsx
const {
    data: mintData,
    isError: mintError,
    isLoading,
    writeAsync: mint,
  } = useContractWrite(useCoinageContract(), 'caucusMint', {
    onError(error) {
      console.log('onError', error);
    },
    onSuccess(data) {
      console.log('Success', data)
    },

  });
  const { data: waitData, error: waitError } = useWaitForTransaction({
    wait: mintData?.wait,
    hash: mintData?.hash,
    onSettled(data, error) {
      console.log('Use Wait on settled', { data, error })
    },
    onError(error) {
      console.log('Use Wait on error', { error })
    },
    onSuccess(data) {
      console.log('Use Wait on succes', { data })
    },
  });
```

<img width="700" alt="image" src="https://user-images.githubusercontent.com/8649366/167318384-abc4c24a-2f48-4a27-95cd-f59c16eeb375.png">

**Comment by isaac-martin** [comment]
Still can't figure this out if anyone has any ideas

**Reply by tmm** [reply]
`onError` is only called if waiting for transaction throws an error. If you want to check if the transaction has failed, use the `data.status === 0` (see ethers [`TransactionReceipt`](https://docs.ethers.io/v5/api/providers/types/#providers-TransactionReceipt) for more info).

**Reply by Ding-Fan** [reply]
> `onError` is only called if waiting for transaction throws an error. If you want to check if the transaction has failed, use the `data.status === 0` (see ethers [`TransactionReceipt`](https://docs.ethers.io/v5/api/providers/types/#providers-TransactionReceipt) for more info).

saved my day :sob: 
the document of ethers js is too difficult to search, and no comment in the `node_modules/@ethersproject/abstract-provider/src.ts/index.ts`
